elessar/pages/Login/login.tsx

```diff
@@ -1,7 +1,116 @@
-import React, { Component, createRef } from 'react';
-import { View, Keyboard, StyleSheet, TouchableOpacity, Button, Text, TextInput} from 'react-native';
+import React, { Component, useState, useEffect, Props }  from 'react';
+import { View, Keyboard, StyleSheet, TouchableOpacity, Text, TextInput} from 'react-native';
 
 
+
+interface LoginState {
+    email: string,
+    message: string,
+    password:string,
+    status: boolean,
+  }
+
+export default class Login extends Component<Props,LoginState>{
+    constructor(props) {
+        super(props);
+        this.state = {
+            email: '',
+            message: '',
+            password: '',
+            status: false,
+        }
+        this.validate_fields = this.validate_fields.bind(this)
+    }
+    
+
+
+    validate_fields(){
```

Comment:
Usamos arrow functions, as arrow functions tem o escopo léxico, ou seja, não altera o escopo. Com isso nao precisamos executar o bind, já que mantes o escopo quando acessamos o this dentro de uma arrow function acessamos o escopo que a função se encontra.
```suggestion
    }

    validate_fields = () => {
```